File: bank-service/src/main/java/rs/edu/raf/dto/ForeignCurrencyBankAccountCreateDTO.java
package rs.edu.raf.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PositiveOrZero;

import java.math.BigDecimal;

public record ForeignCurrencyBankAccountCreateDTO (
        @NotNull Long owner,
        @NotNull Long currencyId,
        @NotNull @PositiveOrZero BigDecimal maintenancePrice
) { }

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/dto/ForeignCurrencyBankAccountCreateDTO.kt
package org.example.bankservicekotlin.dto

import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.PositiveOrZero
import java.math.BigDecimal

data class ForeignCurrencyBankAccountCreateDTO(
    @NotNull val JMBG: String,
    @NotNull val currencyId: Long,
    @NotNull @PositiveOrZero val maintenancePrice: BigDecimal
)
